client/ayon_flame/api/plugin.py

```diff
@@ -621,19 +630,45 @@ def _convert_to_marker_data(self):
                 `tag_hierarchy_data` will be set only once for every
                 clip which is not hero clip.
                 """
-                _hero_data = deepcopy(hero_data)
-                _hero_data.update({"heroTrack": False})
+                _distrib_data = deepcopy(hero_data)
+                _distrib_data.update({"heroTrack": False})
                 if _in <= self.clip_in and _out >= self.clip_out:
                     data_product_name = hero_data["productName"]
-                    # add track index in case duplicity of names in hero data
-                    if self.product_name in data_product_name:
-                        _hero_data["productName"] = self.product_name + str(
-                            self.track_index)
+                    used_names_list = self.vertical_clip_used.setdefault(
+                        data_product_name, [])
+                    self.log.debug(
+                        f">> used_names_list: {used_names_list}"
+                    )
+                    clip_product_name = self.product_name
+                    self.log.debug(
+                        f">> clip_product_name: {clip_product_name}")
+
                     # in case track name and product name is the same then add
                     if self.base_product_name == self.track_name:
-                        _hero_data["productName"] = self.product_name
+                        clip_product_name = self.product_name
+
+                    # add track index in case duplicity of names in hero data
```

Comment:
@jakubjezek001 thins is outdated in code, but still relevant.

Reply:
Yes those product names are changed to be `suffixed` only in case the duplicity is found. So I would say we need to add it this way. Or is there some better way to do it. We do not need the suffix if there is not any duplicity.

Reply:
This is dangerous, you should either always add all suffixes, or handle it differently. You have 2 conditions that both can add different numeric suffix. So you can't tell if previously used product name has track index or "cs index".
```python
# Current code

# This adds track index if clip product name is in "data product name" (whatever that is?)
# - shouldn't this be 'if clip_product_name == data_product_name:'?
if clip_product_name in data_product_name:
    clip_product_name = f"{clip_product_name}{self.track_index}"

# Here you check if product name is already user, but you don't validate if the new name is also already used
#     so you can create already used product name here
if clip_product_name in used_names_list:
    clip_product_name = f"{clip_product_name}{self.cs_index}"
```

Reply:
Yes the result can be unexpected and imlicit but is it really dangerous?

Reply:
You tell me, I'm pointing to potential danger. All these conditions are there to avoid duplication of product name, but they may cause a different duplication 🙂 .

Reply:
yes there is the section with track index where the condition should not be `in` but `==`. Thanks.